### ISSUE TITLE
use "beforeExecuteRoute" instead of "beforeDispatch" in the Security plugin

### ISSUE THREAD

**Issue by vladbalmos** [phalcon/invo#27]
The Security plugin should use "beforeExecuteRoute" instead of the "beforeDispatch" event in order to check the ACL. 

In case the user tries to access a controller/action that doesn't exist the ACL will report that the client is not allowed to access the said resource, since it wasn't listed in the ACL.

Example:
role: Users -> GET /companies/invalid-action; ACL will report denied, since "invalid-action" wasn't added to the list

"beforeExecuteRoute" event is triggered only after the dispatcher knows that the controller/action exists and will allow 404  errors to be handled.

**Comment by chiWahWong** [comment]
seem like directly changing it to "beforeExecuteRoute" will skip the authentication.

**Comment by sergeyklay** [comment]
Fixed in the `0.6.x` branch. Feel free to open a new issue if the problem appears again. Thank you for contributing.
